File: frontend/src/services/fuelRate/fuelService.js
import privateApi from "../privateApi";

export const getAllFuels = () =>
  privateApi.get("/fuels");

export const addFuel = (data) =>
  privateApi.post("/fuels", data);

export const updateFuelStatus = (id, status) =>
  privateApi.patch(`/fuels/${id}/status?status=${status}`);

export const deleteFuel = (fuelId) => {
  return privateApi.delete(`/fuels/${fuelId}`);
};

File: backend/src/main/java/com/example/backend/services/fuel_rate/FuelService.java
package com.example.backend.services.fuel_rate;



import java.util.List;

import com.example.backend.dto.fuel_rate.FuelRequest;
import com.example.backend.dto.fuel_rate.FuelResponse;



public interface FuelService {

    FuelResponse addFuel(FuelRequest request);

    List<FuelResponse> getAllFuels();

    FuelResponse updateFuelStatus(Long fuelId, String status);
    
    void deleteFuel(Long fuelId);

}